src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5520,11 +5520,12 @@ void makeAndAdjustReceiverSlot(BoundExpression receiver)
             {
                 if (!wasTargetTyped)
                 {
-                    // This can happen when we're inferring the return type of a lambda. For this case, we don't need to do any work,
+                    // This can happen when we're inferring the return type of a lambda or visiting a suppressed node like
+                    // BoundConvertedTupleLiteral.SourceTuple. For these cases, we don't need to do any work,
                     // the unconverted conditional operator can't contribute info. The conversion that should be on top of this
                     // can add or remove nullability, and nested nodes aren't being publicly exposed by the semantic model.
                     Debug.Assert(node is BoundUnconvertedConditionalOperator);
-                    Debug.Assert(_returnTypesOpt is not null);
+                    Debug.Assert(_returnTypesOpt is not null || _disableDiagnostics);
```

Comment:
>Debug.Assert(_returnTypesOpt is not null || _disableDiagnostics); [](http://example.com/codeflow?start=20&length=65)

I understand that this change disables the assert for the scenario, but I am not sure why it is the right change. Do we understand why we are getting here? For example, it is not obvious to me that we expect to analyze an unconverted conditional operator for a success scenario not involving lambda re-inference.

Reply:
> Do we understand why we are getting here?

Yes, because we visit the unconverted tuple here:

https://github.com/dotnet/roslyn/blob/2d503182bc7760b3aa5c7a8d5a8ee662dcc38852/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L7588-L7590

> it is not obvious to me that we expect to analyze an unconverted conditional operator for a success scenario not involving lambda re-inference.

I'm not sure how we could avoid that. If we want to visit the unconverted source tuple, it seems we have to visit everything inside (including unconverted conditional operator).

Reply:
Disabling assert for this scenario feels reasonable. However, using ```_disableDiagnostics``` for that feels like a too much of a relaxation. I am curious what @cston and @jcouv think about this. Is it worth keeping the assert at all?

Reply:
I think we could remove the assert entirely.